packages/react-tinacms-github/src/github-editing-context/TinacmsGithubProvider.tsx

```diff
@@ -16,64 +16,51 @@ limitations under the License.
 
 */
 
-import React, { useState, useEffect, useCallback } from 'react'
+import React, { useState, useEffect } from 'react'
 import { useCMS } from 'tinacms'
 import GithubErrorModal from '../github-error/GithubErrorModal'
 import GithubAuthModal from './GithubAuthModal'
 import { GithubEditingContext } from './GithubEditingContext'
 import { useGithubEditing } from './useGithubEditing'
-import { authenticate as githubAuthenticate } from '../github-auth'
 import { GithubClient } from '../github-client'
 
 interface ProviderProps {
   children: any
-  clientId: string
-  authCallbackRoute: string
   editMode: boolean
   enterEditMode: () => void
   exitEditMode: () => void
   error?: any
 }
 
-interface AuthState {
-  authenticated: true
-  forkValid: true
-}
+type AuthStep = null | 'authenticate' | 'createFork'
 
 export const TinacmsGithubProvider = ({
   children,
   editMode,
   enterEditMode,
   exitEditMode,
-  authCallbackRoute,
-  clientId,
   error: previewError,
 }: ProviderProps) => {
   const [error, setError] = useState<any>(null)
   const cms = useCMS()
   const github: GithubClient = cms.api.github
-  const [authorizingStatus, setAuthorizingStatus] = useState<AuthState | null>(
-    null
-  )
+  const [authStep, setAuthStep] = useState<AuthStep>(null)
 
   const tryEnterEditMode = async () => {
     const authenticated =
-      authorizingStatus?.authenticated || (await github.getUser())
-    const forkValid = authorizingStatus?.forkValid || (await github.getBranch())
-
-    if (authenticated && forkValid) {
-      enterEditMode()
-    } else {
-      setAuthorizingStatus({
-        authenticated,
-        forkValid,
-      })
+      authStep === 'authenticate' || (await github.getUser())
```

Comment:
Don't really understand this check. If we are on the 'authenticate' step, why should `authenticated = true` ?

Reply:
Ahh, actually I guess this behaviour hasn't really changed.